### PR TITLE
Pull request for grid printer

### PR DESCRIPTION
**Pull request by shaynaalice**
Why do I keep creating a swap file?

### CODE REVIEW

File: students/shayna_andersonhill/grid_printer_exercise/grid_printer.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+print(line, space*8, line, space*8, line)
+print(line, space*8, line, space*8, line)
+print(line, space*8, line, space*8, line)
+print(line, space*8, line, space*8, line)
+print(plus, (minus+space)*4,plus, (minus+space)*4,plus)
+print(line, space*8, line, space*8, line)
+print(line, space*8, line, space*8, line)
+print(line, space*8, line, space*8, line)
+print(line, space*8, line, space*8, line)
+print(plus, (minus+space)*4,plus, (minus+space)*4,plus)
```
Comment (mdgregor) [inline students/shayna_andersonhill/grid_printer_exercise/grid_printer.py]: Can you think of how you might write this with a `for` loop? HINT: nested for loops :)

File: students/shayna_andersonhill/grid_printer_exercise/grid_printer.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#Create function to print grid customized by input value n
+
+def horizontals(n, columns=2):
+    print(plus, (minus+space)*(n-1) + minus, plus, (minus + space)*(n-1)+
```
Comment (mdgregor) [inline students/shayna_andersonhill/grid_printer_exercise/grid_printer.py]: try to be consistant with the spacing between operators:
```
print(plus, (minus + space) * (n-1) + minus, plus, (minus + space) * (n-1) +
```

File: students/shayna_andersonhill/grid_printer_exercise/grid_printer.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        print(line, space*((n*2)-1), line, space*((n*2)-1), line)
+
+print('Part 2:')
+
+def print_standard_grid(n, size = 2):
+    horizontals(n)
+    verticals(n)
+    horizontals(n)
+    verticals(n)
+    horizontals(n)
```
Comment (mdgregor) [inline students/shayna_andersonhill/grid_printer_exercise/grid_printer.py]: Can you convert these into a nested `for` loop as well?

File: students/shayna_andersonhill/grid_printer_exercise/grid_printer.py (hunk 1 above)
Comment (shaynaalice) [reply students/shayna_andersonhill/grid_printer_exercise/grid_printer.py]: Can you review my nested for loop solution. Would it be better if I found a way to include the last print statement as part of the loop?

File: students/shayna_andersonhill/lesson02/grid_printer.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+space = ' '
+line = '|'
+
+#Naive Answer
+print('Part One')
+for i in range(1,3):
+    print(plus, (minus+space)*4, plus, (minus+space)*4, plus)
+    for i in range(1,5):
+        print(line, space*8, line, space*8, line)
+print(plus, (minus+space)*4, plus, (minus+space)*4, plus)
```
Comment (mdgregor) [inline students/shayna_andersonhill/lesson02/grid_printer.py]: Nicely done.

File: students/shayna_andersonhill/lesson02/grid_printer.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    print(plus, (minus+space)*(n-1)+minus, plus, (minus+space)*(n-1)+minus, plus)
+
+def verticals(n):
+    for i in range(n):
+        print(line, space*((n*2)-1), line, space*((n*2)-1), line)
+
+print('Part 2:')
+def print_standard_grid(n, size = 2):
+    for i in range(1,3):
+        horizontals(n)
```
Comment (mdgregor) [inline students/shayna_andersonhill/lesson02/grid_printer.py]: So in this instance, you didn't need to call a function to execute a single line.  better to just keep the original line even if you copy paste it to use it elsewhere.  We use functions primaries for multiple statements (with exceptions of course).

File: students/shayna_andersonhill/lesson02/grid_printer.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+units = int(input('How many units?'))
+rows = int(input('How many rows?'))
+columns = int(input('How many columns?'))
+
+def column(columns, units):
+    for i in range(columns):
+        print(plus, (minus+space)*(units-1)+minus, end=' ')
+    print(plus)
+
+def row(rows, units, columns):
```
Comment (mdgregor) [inline students/shayna_andersonhill/lesson02/grid_printer.py]: You feed `rows` into this function but never use it.

**Pull request merged**